homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +124,45 @@ async def async_setup_entry(
+                    ].cost.currency_unit
+                entities.append(OVOEnergySensor(coordinator, description, client))
 
     async_add_entities(entities, True)
 
 
 class OVOEnergySensor(OVOEnergyDeviceEntity, SensorEntity):
-    """Defines a OVO Energy sensor."""
+    """Define a OVO Energy sensor."""
 
-    _attr_state_class = STATE_CLASS_TOTAL_INCREASING
+    coordinator: DataUpdateCoordinator
+    entity_description: OVOEnergySensorEntityDescription
 
     def __init__(
         self,
         coordinator: DataUpdateCoordinator,
+        description: OVOEnergySensorEntityDescription,
         client: OVOEnergy,
-        key: str,
-        name: str,
-        icon: str,
-        device_class: str | None,
-        unit_of_measurement: str | None,
     ) -> None:
-        """Initialize OVO Energy sensor."""
-        self._attr_device_class = device_class
-        self._unit_of_measurement = unit_of_measurement
-
-        super().__init__(coordinator, client, key, name, icon)
-
-    @property
-    def native_unit_of_measurement(self) -> str | None:
-        """Return the unit this state is expressed in."""
-        return self._unit_of_measurement
-
-
-class OVOEnergyLastElectricityReading(OVOEnergySensor):
-    """Defines a OVO Energy last reading sensor."""
-
-    def __init__(self, coordinator: DataUpdateCoordinator, client: OVOEnergy) -> None:
-        """Initialize OVO Energy sensor."""
-
-        super().__init__(
-            coordinator,
-            client,
-            f"{client.account_id}_last_electricity_reading",
-            "OVO Last Electricity Reading",
-            "mdi:flash",
-            DEVICE_CLASS_ENERGY,
-            "kWh",
-        )
-
-    @property
-    def native_value(self) -> str:
-        """Return the state of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return usage.electricity[-1].consumption
-
-    @property
-    def extra_state_attributes(self) -> object:
-        """Return the attributes of the sensor."""
-        usage: OVODailyUsage = self.coordinator.data
-        if usage is None or not usage.electricity:
-            return None
-        return {
-            "start_time": usage.electricity[-1].interval.start,
-            "end_time": usage.electricity[-1].interval.end,
-        }
-
-
-class OVOEnergyLastGasReading(OVOEnergySensor):
-    """Defines a OVO Energy last reading sensor."""
-
-    def __init__(self, coordinator: DataUpdateCoordinator, client: OVOEnergy) -> None:
-        """Initialize OVO Energy sensor."""
-
+        """Initialize."""
         super().__init__(
             coordinator,
             client,
-            f"{DOMAIN}_{client.account_id}_last_gas_reading",
-            "OVO Last Gas Reading",
-            "mdi:gas-cylinder",
-            DEVICE_CLASS_ENERGY,
-            "kWh",
+            f"{DOMAIN}_{client.account_id}_{description.key}",
```

Comment:
This is only used for the unique ID, you can just set that here instead of passing this to key in a parent class